File: examples/typegraphs/metagen/py/remix.py
from .fdk import handler_remix_track, Idv3, Ctx


# the following decorator makes sure your function
# adheres to the function types from the typegraph
@handler_remix_track
def remix_track(inp: Idv3, _cx: Ctx) -> Idv3:
    return {
        "title": f"{inp['title']} (Remix)",
        "artist": f"{inp['artist']} + DJ Cloud",
        "release_time": inp["release_time"],
        "mp3_url": "https://mp3.url/remix1",
    }

File: examples/typegraphs/metagen/ts/remix.ts
import type { Ctx, Idv3, RemixTrackHandler } from "./fdk.ts";

// the name of the export must match the one referred int he typegraph
export const remix_track: RemixTrackHandler = (inp, cx: Ctx) => {
  const out: Idv3 = {
    title: `${inp.title} (Remix)`,
    artist: `${inp.artist} + DJ Cloud`,
    releaseTime: new Date().toISOString(),
    // S3Runtime could be used to really provide this service
    mp3Url: `${cx.meta.url}/get_mp3`,
  };
  return out;
};
